src/it/scala/net/snowflake/spark/snowflake/ShareConnectionSuite.scala

```diff
@@ -423,4 +425,26 @@ class ShareConnectionSuite extends IntegrationSuiteBase {
     assert(result2.getString(1).equals("ABORT_DETACHED_QUERY"))
     assert(result2.getBoolean(2))
   }
+
+  test("use provided connections") {
+    val conn1 =
+      ServerConnection
+        .getServerConnection(Parameters.mergeParameters(connectorOptions)).jdbcConnection
+    val id1 = ServerConnection.providedConnections.register(conn1)
```

Comment:
This is an internal API can be accessed from snowflake package only. We may open it to public in feature if necessary.